scraper.py

```diff
@@ -367,17 +412,20 @@ def get_book(self, selected_title_link: str, download_path: str):
                     chapter_table_close.click()
                     time.sleep(1)
 
+                    current_location = convert_metadata.to_seconds(timeline_current_time.get_attribute("textContent"))
+                    current_chapter = self.chapter_containing(current_location)
+                    current_chapter_start = self.chapter_seconds[current_chapter]
+
                     if upper_bound == expected_duration:
                         # Can get to the end of the audio with one more click.
                         chapter_next.click()
                         time.sleep(1)
-
+                    elif current_chapter == desired_chapter and current_chapter_start != current_location:
+                        # This seems to be the result of navigating away from a chapter.
+                        chapter_previous.click()
+                        time.sleep(1)
```

Comment:
Turns out Overdrive might remember where you are per-chapter: when hopping by menu I'd sometimes find odd offsets. This takes care of that.